addok/http/base.py

```diff
@@ -139,8 +139,11 @@ def on_get(self, req, resp, **kwargs):
         if lon and lat:
             center = (lon, lat)
         filters = self.match_filters(req)
-        results = search(query, limit=limit, autocomplete=autocomplete,
-                         lat=lat, lon=lon, **filters)
+        try:
+            results = search(query, limit=limit, autocomplete=autocomplete,
+                             lat=lat, lon=lon, **filters)
+        except ValueError as e:
+            raise falcon.HTTPBadRequest(str(e))
```

Comment:
I finally used a 400 (not 413), so we can keep this check generic: now any place in the process can raise a ValueError, and this will be properly caught.